frontend/packages/dev-console/src/components/topology/topology-utils.ts

```diff
@@ -0,0 +1,436 @@
+    const deploymentsLabels = _.get(configRes, properties) || {};
+    return !!deploymentsLabels['serving.knative.dev/configuration'];
+  }
+
+  /**
+   * Get all the pods from a replication controller or a replicaset.
+   * @param replicationController
+   */
+  private getPods(replicationController: ResourceProps) {
+    const dcPodsData = _.filter(this.resources.pods.data, (pod) => {
+      return _.some(_.get(pod, 'metadata.ownerReferences'), {
+        uid: _.get(replicationController, 'metadata.uid'),
+        controller: true,
+      });
+    });
+    if (
+      dcPodsData &&
+      !dcPodsData.length &&
+      this.isKnativeServing(replicationController, 'metadata.labels')
+    ) {
+      return [
+        {
+          ..._.pick(replicationController, 'metadata', 'status', 'spec'),
+          status: { phase: 'Scaled To 0' },
+        },
+      ];
+    }
+    return dcPodsData;
+  }
+
+  /**
+   * fetches all the replication controllers from the deployment
+   * @param deploymentConfig
+   * @param targetDeployment 'deployments' || 'deploymentConfigs'
+   */
+  private getReplicationController(
+    deploymentConfig: ResourceProps,
+    targetDeployment: string,
+  ): ResourceProps {
+    // Get the current replication controller or replicaset
+    const targetReplicationControllersKind = this.deploymentKindMap[targetDeployment].rcKind;
+    const replicationControllers = this.deploymentKindMap[targetDeployment].rController;
+    const dcUID = _.get(deploymentConfig, 'metadata.uid');
+
+    const rControllers = _.filter(
+      this.resources[replicationControllers].data,
+      (replicationController) => {
+        return _.some(_.get(replicationController, 'metadata.ownerReferences'), {
+          uid: dcUID,
+          controller: true,
+        });
+      },
+    );
+    const sortedControllers = this.sortByDeploymentVersion(rControllers, true);
+    return _.merge(_.head(sortedControllers), {
+      kind: targetReplicationControllersKind,
+    });
+  }
+
+  /**
+   * create graph data from the deploymentconfig.
+   * @param deploymentConfig
+   */
+  private createGraphData(deploymentConfig) {
+    // Current Node data
+    const { metadata } = deploymentConfig;
+    const currentNode = {
+      id: metadata.uid,
+      type: 'workload',
+      name: (metadata.labels && metadata.labels['app.openshift.io/instance']) || metadata.name,
+    };
+
+    if (!_.some(this.topologyData.graph.nodes, { id: currentNode.id })) {
+      // add the node to graph
+      this.topologyData.graph.nodes.push(currentNode);
+      const labels = _.get(deploymentConfig, 'metadata.labels');
+      const annotations = _.get(deploymentConfig, 'metadata.annotations');
+      let edges = [];
+      const totalDeployments = _.cloneDeep(
+        _.concat(this.resources.deploymentConfigs.data, this.resources.deployments.data),
+      );
+      // find and add the edges for a node
+      if (_.has(annotations, 'app.openshift.io/connects-to')) {
```

Comment:
Does this work? I'd think it would be have to be the following due to special characters

```suggestion
      if (_.has(annotations, ['app.openshift.io/connects-to'])) {
```

Reply:
Ya it works. But good catch as this param is expected to be a path. I'll update.

Reply:
Updated.